americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+        ax = plt.axes()
+
+    # Sets the colormap and egdecolor, if necessary
+    if colormap is None:
+        colormap = np.array([[0.5, 0.5, 0.5]]*len(height))
+    elif isinstance(colormap, str):
+        colormap = segment_colormap(colormap, len(height))
+
+    if match_colors:
+        edgecolors = colormap
+    else:
+        edgecolors = np.array([[0, 0, 0]]*len(height))
+
+    # Gets an axis instance
+    if ax is None:
+        ax = plt.axes()
+
+    # Gets the xposition, for errorbars
+    xpos = np.arange(0, len(height))*interval + interval/2 + offset
+
+    xleft = np.arange(0, len(height)) * interval + (interval - width)/2 + offset
+    xlims = [0, len(height)*interval]
+
+    if 'xlim' not in kwargs or kwargs['xlim'] is None:
+        kwargs['xlim'] = xlims
+
+    # Plots the errorbars
+    if errors is not None:
+        e_bars = []
+        for idx, err in enumerate(errors):
+            eb = ax.errorbar(x=xpos[idx],
+                             y=height[idx],
+                             yerr=err,
+                             fmt='none',
+                             ecolor=edgecolors[idx, :],
+                             elinewidth=elinewidth,
+                             capthick=ecapwidth)
+        e_bars.append(eb)
+
+    # Plots the bars
+    bars = ax.bar(xleft, height, width=width)
+    for idx, bar in enumerate(bars):
+        bar.set_facecolor(colormap[idx, :])
+        bar.set_edgecolor(edgecolors[idx, :])
+
+    _format_axis(ax)
+
+    return ax, xpos
+
+
+def add_comparison_bars(centers, tops, p_values, ax, space=None,
+    interval=None, lowest=None, factor=5, label_size=10,
+    show_value=True):
+    """Adds p_value bars
+
+    The assumes that comparison bars are being introduced for a
+    group of error bars, assuming a reference and comparison.
+
+    Parameters
+    ----------
+    centers : array-like
+        The center of the barplot.
+    tops : array_like
+        The maximum height of the groups to be compared. There must be a
+        top for each center supplied.
+    p_values : array-like
+        The critical values between the groups in the bars. There should be
+        one less p value than there are bars.
+    ax : matplotlib axis, optional
+        The axis where data should be plotted. If none, a new axis instance
+        will be created.
+    space : float
+        The y-distance between each the comparison line.
+    interval : float
+        The space between the bars
+    lowest : float, optional
+        If mutliple sets of comparison bars are being added, this allows
+        the user to set the position of the lowest bar in the group
+    factor : unsigned int, optional
+        The ones-place value to which values will be rounded. For example,
+        for a `factor` of 5, a value of 0.12 will be rounded to 0.15 and a
+        value of 2.7 will be rounded to 3.0.
+    label_size : unsigned int, optional
+        The font size for displaying signifigance labels
```

Comment:
significance

...gonna stop mentioning this one. suggest doing a search through the code for "signi" and check that way?